File: C/05-MARÇO/15-05/SIMPLE/stack.c
#include <stdio.h>
#include "stack.h"
#define CON 100

int main()
{
    Stack stack = stack_construct();
    for(int i=0; i<CON; i++)
    {
        push(&stack, i);
    }
    
    for(int i=0; i<CON; i++)
    {
        pop(&stack);
        printf("%d\n",peek(&stack));
    }
}

File: 99-JAVA/05 - Maio/O16/Stack.java
public class Stack<T>
{
    // push pop peek size
    // array dinamico
    T[] data;
    Integer size = 0;
    Stack()
    {
        data = (T[])(new Object[10]);
    }
    
    Stack( int capacity )
    {
        data = (T[])(new Object[capacity]);
    }

    public void push(Integer index, T value)
    {


        if ( index >= data.length )
        {
            T[] copy = (T[])( new Object[ index+1 ]);
            for( int i = 0 ; i < size ; i++ )
            {
                copy[i] = this.data[i];
            }
        }
        this.data[index] = value;
        System.err.println(index);
        System.out.println(value);
        System.out.print("\n\n\n");
        this.size++;
    }

    public void pop(){
        this.data[this.size] = null;
        this.size--;
    }

    public T peek()
    {
        return this.data[this.size];
    }

    public Integer getSize()
    {   
        return this.size;    
    }
}
